{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fe716014-6c08-42cc-abe2-e1cb0cac8e38",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'getOption(\"repos\")' replaces Bioconductor standard repositories, see\n",
      "'help(\"repositories\", package = \"BiocManager\")' for details.\n",
      "Replacement repositories:\n",
      "    CRAN: https://cran.r-project.org\n",
      "\n",
      "Bioconductor version 3.18 (BiocManager 1.30.26), R 4.3.3 (2024-02-29)\n",
      "\n",
      "Warning message:\n",
      "\"package(s) not installed when version(s) same as or greater than current; use\n",
      "  `force = TRUE` to re-install: 'GenomicRanges' 'GenomicFeatures'\n",
      "  'TxDb.Hsapiens.UCSC.hg38.knownGene' 'org.Hs.eg.db' 'biomaRt'\n",
      "  'EnsDb.Hsapiens.v86'\"\n",
      "Old packages: 'AnnotationHub', 'BSgenome', 'BiocFileCache', 'Biostrings',\n",
      "  'GOSemSim', 'GenomeInfoDb', 'GenomicAlignments', 'HDF5Array', 'MBSP',\n",
      "  'MOFA2', 'RSQLite', 'RcppArmadillo', 'RcppParallel', 'Rfast', 'Rhdf5lib',\n",
      "  'Rhtslib', 'S4Arrays', 'SparseArray', 'XML', 'basilisk', 'beachmat',\n",
      "  'bigsnpr', 'caret', 'crosstalk', 'curl', 'dqrng', 'evaluate', 'gert',\n",
      "  'ggiraph', 'ggpattern', 'ggraph', 'ggridges', 'ggtree', 'gower',\n",
      "  'gwasrapidd', 'hardhat', 'later', 'leidenbase', 'patchwork', 'pgenlibr',\n",
      "  'psichomics', 'restfulr', 'rprojroot', 's2', 'sf', 'shadowtext', 'susieR',\n",
      "  'tidygraph', 'unigd', 'xml2', 'zlibbioc'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "BiocManager::install(c(\"GenomicRanges\", \"GenomicFeatures\", \n",
    "                      \"TxDb.Hsapiens.UCSC.hg38.knownGene\", \n",
    "                      \"org.Hs.eg.db\", \"biomaRt\", \"EnsDb.Hsapiens.v86\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "95f4e280-e3e8-4e8e-a36f-84523b3463a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- \u001b[1mAttaching core tidyverse packages\u001b[22m ------------------------ tidyverse 2.0.0 --\n",
      "\u001b[32mv\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.4     \u001b[32mv\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.5\n",
      "\u001b[32mv\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32mv\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.1\n",
      "\u001b[32mv\u001b[39m \u001b[34mggplot2  \u001b[39m 3.5.2     \u001b[32mv\u001b[39m \u001b[34mtibble   \u001b[39m 3.3.0\n",
      "\u001b[32mv\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.4     \u001b[32mv\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.1\n",
      "\u001b[32mv\u001b[39m \u001b[34mpurrr    \u001b[39m 1.1.0     \n",
      "-- \u001b[1mConflicts\u001b[22m ------------------------------------------ tidyverse_conflicts() --\n",
      "\u001b[31mx\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31mx\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mi\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\n",
      "Attaching package: 'data.table'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:lubridate':\n",
      "\n",
      "    hour, isoweek, mday, minute, month, quarter, second, wday, week,\n",
      "    yday, year\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from 'package:purrr':\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5afec9ca-97ae-47af-a5af-9bed1002c686",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:lubridate':\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:lubridate':\n",
      "\n",
      "    second, second<-\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from 'package:tidyr':\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "The following object is masked from 'package:utils':\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:data.table':\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from 'package:lubridate':\n",
      "\n",
      "    %within%\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "The following object is masked from 'package:purrr':\n",
      "\n",
      "    reduce\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'AnnotationDbi'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: ensembldb\n",
      "\n",
      "Loading required package: AnnotationFilter\n",
      "\n",
      "\n",
      "Attaching package: 'ensembldb'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:dplyr':\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    filter\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "library(GenomicRanges)\n",
    "library(GenomicFeatures)\n",
    "library(TxDb.Hsapiens.UCSC.hg38.knownGene)\n",
    "library(org.Hs.eg.db)\n",
    "library(EnsDb.Hsapiens.v86)\n",
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3b080271-b553-485e-9877-8e5163859991",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_cb = fread(\"~/project/image_QTL/interested_genes/CB_M1/Image_AD_aging_PD_meta_colocboost_export_filtered_ucos.bed\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5d246638-18ca-4ad1-a952-02079aa1cc14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 14</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>a1</th><th scope=col>a2</th><th scope=col>variant_ID</th><th scope=col>region_ID</th><th scope=col>event_ID</th><th scope=col>cos_ID</th><th scope=col>vcp</th><th scope=col>cos_npc</th><th scope=col>min_npc_outcome</th><th scope=col>coef</th><th scope=col>z</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>161189357</td><td>161189357</td><td>T</td><td>C</td><td>chr1:161189357:C:T</td><td>chr1_158933969_162352348</td><td>AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        </td><td>chr1_158933969_162352348:cos1:y7_y10 </td><td>0.148633848</td><td>1</td><td>1</td><td>-0.00237822429413167;-0.00126943889431771</td><td>-5.66666666666667;-5.39920492844837</td></tr>\n",
       "\t<tr><td>chr1</td><td>161215268</td><td>161215268</td><td>T</td><td>G</td><td>chr1:161215268:G:T</td><td>chr1_158933969_162352348</td><td>AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        </td><td>chr1_158933969_162352348:cos1:y7_y10 </td><td>0.213683842</td><td>1</td><td>1</td><td>-0.00314917325108774;-0.0015555119117074 </td><td>-5.72826086956522;-5.44139840072206</td></tr>\n",
       "\t<tr><td>chr1</td><td>161217875</td><td>161217875</td><td>T</td><td>C</td><td>chr1:161217875:C:T</td><td>chr1_158933969_162352348</td><td>AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        </td><td>chr1_158933969_162352348:cos1:y7_y10 </td><td>0.630037093</td><td>1</td><td>1</td><td>-0.0035129752975932;-0.00589550094118934 </td><td>-5.71276595744681;-5.78719041195195</td></tr>\n",
       "\t<tr><td>chr1</td><td>200905600</td><td>200905600</td><td>G</td><td>A</td><td>chr1:200905600:A:G</td><td>chr1_200496446_202852808</td><td>AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021</td><td>chr1_200496446_202852808:cos1:y10_y11</td><td>0.007671881</td><td>1</td><td>1</td><td>9.234320806912e-05;0.000139149039132946  </td><td>4.53950141738719;4.29121674501588  </td></tr>\n",
       "\t<tr><td>chr1</td><td>200905967</td><td>200905967</td><td>C</td><td>T</td><td>chr1:200905967:T:C</td><td>chr1_200496446_202852808</td><td>AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021</td><td>chr1_200496446_202852808:cos1:y10_y11</td><td>0.008368034</td><td>1</td><td>1</td><td>0.000104268954830228;0.000138364785535695</td><td>4.58025882916672;4.28441576043348  </td></tr>\n",
       "\t<tr><td>chr1</td><td>200906114</td><td>200906114</td><td>A</td><td>T</td><td>chr1:200906114:T:A</td><td>chr1_200496446_202852808</td><td>AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021</td><td>chr1_200496446_202852808:cos1:y10_y11</td><td>0.008713429</td><td>1</td><td>1</td><td>0.000107208760913441;0.000142027094052532</td><td>4.58984897079625;4.29387534466063  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 14\n",
       "\\begin{tabular}{llllllllllllll}\n",
       " \\#chr & start & end & a1 & a2 & variant\\_ID & region\\_ID & event\\_ID & cos\\_ID & vcp & cos\\_npc & min\\_npc\\_outcome & coef & z\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1 & 161189357 & 161189357 & T & C & chr1:161189357:C:T & chr1\\_158933969\\_162352348 & AD\\_Bellenguez; AD\\_Wightman\\_Excluding23andMe\\_2021         & chr1\\_158933969\\_162352348:cos1:y7\\_y10  & 0.148633848 & 1 & 1 & -0.00237822429413167;-0.00126943889431771 & -5.66666666666667;-5.39920492844837\\\\\n",
       "\t chr1 & 161215268 & 161215268 & T & G & chr1:161215268:G:T & chr1\\_158933969\\_162352348 & AD\\_Bellenguez; AD\\_Wightman\\_Excluding23andMe\\_2021         & chr1\\_158933969\\_162352348:cos1:y7\\_y10  & 0.213683842 & 1 & 1 & -0.00314917325108774;-0.0015555119117074  & -5.72826086956522;-5.44139840072206\\\\\n",
       "\t chr1 & 161217875 & 161217875 & T & C & chr1:161217875:C:T & chr1\\_158933969\\_162352348 & AD\\_Bellenguez; AD\\_Wightman\\_Excluding23andMe\\_2021         & chr1\\_158933969\\_162352348:cos1:y7\\_y10  & 0.630037093 & 1 & 1 & -0.0035129752975932;-0.00589550094118934  & -5.71276595744681;-5.78719041195195\\\\\n",
       "\t chr1 & 200905600 & 200905600 & G & A & chr1:200905600:A:G & chr1\\_200496446\\_202852808 & AD\\_Wightman\\_Full\\_2021; AD\\_Wightman\\_Excluding23andMe\\_2021 & chr1\\_200496446\\_202852808:cos1:y10\\_y11 & 0.007671881 & 1 & 1 & 9.234320806912e-05;0.000139149039132946   & 4.53950141738719;4.29121674501588  \\\\\n",
       "\t chr1 & 200905967 & 200905967 & C & T & chr1:200905967:T:C & chr1\\_200496446\\_202852808 & AD\\_Wightman\\_Full\\_2021; AD\\_Wightman\\_Excluding23andMe\\_2021 & chr1\\_200496446\\_202852808:cos1:y10\\_y11 & 0.008368034 & 1 & 1 & 0.000104268954830228;0.000138364785535695 & 4.58025882916672;4.28441576043348  \\\\\n",
       "\t chr1 & 200906114 & 200906114 & A & T & chr1:200906114:T:A & chr1\\_200496446\\_202852808 & AD\\_Wightman\\_Full\\_2021; AD\\_Wightman\\_Excluding23andMe\\_2021 & chr1\\_200496446\\_202852808:cos1:y10\\_y11 & 0.008713429 & 1 & 1 & 0.000107208760913441;0.000142027094052532 & 4.58984897079625;4.29387534466063  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 14\n",
       "\n",
       "| #chr &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | a1 &lt;chr&gt; | a2 &lt;chr&gt; | variant_ID &lt;chr&gt; | region_ID &lt;chr&gt; | event_ID &lt;chr&gt; | cos_ID &lt;chr&gt; | vcp &lt;dbl&gt; | cos_npc &lt;dbl&gt; | min_npc_outcome &lt;dbl&gt; | coef &lt;chr&gt; | z &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 161189357 | 161189357 | T | C | chr1:161189357:C:T | chr1_158933969_162352348 | AD_Bellenguez; AD_Wightman_Excluding23andMe_2021         | chr1_158933969_162352348:cos1:y7_y10  | 0.148633848 | 1 | 1 | -0.00237822429413167;-0.00126943889431771 | -5.66666666666667;-5.39920492844837 |\n",
       "| chr1 | 161215268 | 161215268 | T | G | chr1:161215268:G:T | chr1_158933969_162352348 | AD_Bellenguez; AD_Wightman_Excluding23andMe_2021         | chr1_158933969_162352348:cos1:y7_y10  | 0.213683842 | 1 | 1 | -0.00314917325108774;-0.0015555119117074  | -5.72826086956522;-5.44139840072206 |\n",
       "| chr1 | 161217875 | 161217875 | T | C | chr1:161217875:C:T | chr1_158933969_162352348 | AD_Bellenguez; AD_Wightman_Excluding23andMe_2021         | chr1_158933969_162352348:cos1:y7_y10  | 0.630037093 | 1 | 1 | -0.0035129752975932;-0.00589550094118934  | -5.71276595744681;-5.78719041195195 |\n",
       "| chr1 | 200905600 | 200905600 | G | A | chr1:200905600:A:G | chr1_200496446_202852808 | AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021 | chr1_200496446_202852808:cos1:y10_y11 | 0.007671881 | 1 | 1 | 9.234320806912e-05;0.000139149039132946   | 4.53950141738719;4.29121674501588   |\n",
       "| chr1 | 200905967 | 200905967 | C | T | chr1:200905967:T:C | chr1_200496446_202852808 | AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021 | chr1_200496446_202852808:cos1:y10_y11 | 0.008368034 | 1 | 1 | 0.000104268954830228;0.000138364785535695 | 4.58025882916672;4.28441576043348   |\n",
       "| chr1 | 200906114 | 200906114 | A | T | chr1:200906114:T:A | chr1_200496446_202852808 | AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021 | chr1_200496446_202852808:cos1:y10_y11 | 0.008713429 | 1 | 1 | 0.000107208760913441;0.000142027094052532 | 4.58984897079625;4.29387534466063   |\n",
       "\n"
      ],
      "text/plain": [
       "  #chr start     end       a1 a2 variant_ID         region_ID               \n",
       "1 chr1 161189357 161189357 T  C  chr1:161189357:C:T chr1_158933969_162352348\n",
       "2 chr1 161215268 161215268 T  G  chr1:161215268:G:T chr1_158933969_162352348\n",
       "3 chr1 161217875 161217875 T  C  chr1:161217875:C:T chr1_158933969_162352348\n",
       "4 chr1 200905600 200905600 G  A  chr1:200905600:A:G chr1_200496446_202852808\n",
       "5 chr1 200905967 200905967 C  T  chr1:200905967:T:C chr1_200496446_202852808\n",
       "6 chr1 200906114 200906114 A  T  chr1:200906114:T:A chr1_200496446_202852808\n",
       "  event_ID                                                \n",
       "1 AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        \n",
       "2 AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        \n",
       "3 AD_Bellenguez; AD_Wightman_Excluding23andMe_2021        \n",
       "4 AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021\n",
       "5 AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021\n",
       "6 AD_Wightman_Full_2021; AD_Wightman_Excluding23andMe_2021\n",
       "  cos_ID                                vcp         cos_npc min_npc_outcome\n",
       "1 chr1_158933969_162352348:cos1:y7_y10  0.148633848 1       1              \n",
       "2 chr1_158933969_162352348:cos1:y7_y10  0.213683842 1       1              \n",
       "3 chr1_158933969_162352348:cos1:y7_y10  0.630037093 1       1              \n",
       "4 chr1_200496446_202852808:cos1:y10_y11 0.007671881 1       1              \n",
       "5 chr1_200496446_202852808:cos1:y10_y11 0.008368034 1       1              \n",
       "6 chr1_200496446_202852808:cos1:y10_y11 0.008713429 1       1              \n",
       "  coef                                      z                                  \n",
       "1 -0.00237822429413167;-0.00126943889431771 -5.66666666666667;-5.39920492844837\n",
       "2 -0.00314917325108774;-0.0015555119117074  -5.72826086956522;-5.44139840072206\n",
       "3 -0.0035129752975932;-0.00589550094118934  -5.71276595744681;-5.78719041195195\n",
       "4 9.234320806912e-05;0.000139149039132946   4.53950141738719;4.29121674501588  \n",
       "5 0.000104268954830228;0.000138364785535695 4.58025882916672;4.28441576043348  \n",
       "6 0.000107208760913441;0.000142027094052532 4.58984897079625;4.29387534466063  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(all_cb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "4d2abc82-f149-4c81-9357-1fc188021c0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "TL1 = all_cb |> dplyr::filter(str_detect(event_ID,\"Telomere_pc1\")) |>  dplyr::select(`#chr`, start,variant_ID) |> dplyr::rename(chr = `#chr`, pos = start, variant_id = variant_ID)\n",
    "mvAge = all_cb |> dplyr::filter(str_detect(event_ID,\"mvage\")) |>  dplyr::select(`#chr`, start,variant_ID) |> dplyr::rename(chr = `#chr`, pos = start, variant_id = variant_ID)\n",
    "PD = all_cb |> dplyr::filter(str_detect(event_ID,\"PD\")) |>  dplyr::select(`#chr`, start,variant_ID) |> dplyr::rename(chr = `#chr`, pos = start, variant_id = variant_ID)\n",
    "Aging = rbind(TL1,mvAge)\n",
    "AD = all_cb |> dplyr::filter(str_detect(event_ID,\"AD_\")) |>  dplyr::select(`#chr`, start,variant_ID) |> dplyr::rename(chr = `#chr`, pos = start, variant_id = variant_ID)\n",
    "DNE = all_cb |> dplyr::filter(str_detect(event_ID,\"image_\")) |>  dplyr::select(`#chr`, start,variant_ID) |> dplyr::rename(chr = `#chr`, pos = start, variant_id = variant_ID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "54dfd31e-a02a-4186-b68b-743c90b53216",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2135 genes were dropped because they have exons located on both strands\n",
      "  of the same reference sequence or on more than one reference sequence,\n",
      "  so cannot be represented by a single genomic range.\n",
      "  Use 'single.strand.genes.only=FALSE' to get all the genes in a\n",
      "  GRangesList object, or use suppressMessages() to suppress this message.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Script loaded successfully!\n",
      "Use run_variant_analysis() to process your variants.\n",
      "The output TSV file will now include ENSG_ID column.\n",
      "EnsDb.Hsapiens.v86 detected - enhanced ENSG mapping will be used by default.\n"
     ]
    }
   ],
   "source": [
    "# Efficient Gene Identification for 20,000+ Variants using TxDb in R\n",
    "\n",
    "# Initialize TxDb\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "# Extract all genes once (for efficiency)\n",
    "all_genes <- genes(txdb)\n",
    "\n",
    "# Function for batch processing many variants\n",
    "find_genes_batch <- function(variants_df, chr_col = \"chr\", pos_col = \"pos\", \n",
    "                            id_col = \"variant_id\", window_size = 500000) {\n",
    "  \n",
    "  cat(\"Processing\", nrow(variants_df), \"variants...\\n\")\n",
    "  \n",
    "  # Convert variants to GRanges\n",
    "  variants_gr <- GRanges(\n",
    "    seqnames = variants_df[[chr_col]],\n",
    "    ranges = IRanges(start = variants_df[[pos_col]], \n",
    "                    end = variants_df[[pos_col]]),\n",
    "    variant_id = variants_df[[id_col]]\n",
    "  )\n",
    "  \n",
    "  # Expand variants by window size\n",
    "  expanded_variants <- variants_gr + window_size\n",
    "  \n",
    "  # Find overlaps (vectorized operation - very fast)\n",
    "  overlaps <- findOverlaps(expanded_variants, all_genes)\n",
    "  \n",
    "  # Extract results efficiently\n",
    "  variant_hits <- queryHits(overlaps)\n",
    "  gene_hits <- subjectHits(overlaps)\n",
    "  \n",
    "  # Create results data frame\n",
    "  results <- data.frame(\n",
    "    variant_id = mcols(variants_gr)$variant_id[variant_hits],\n",
    "    variant_chr = as.character(seqnames(variants_gr)[variant_hits]),\n",
    "    variant_pos = start(variants_gr)[variant_hits],\n",
    "    gene_id = mcols(all_genes)$gene_id[gene_hits],\n",
    "    gene_chr = as.character(seqnames(all_genes)[gene_hits]),\n",
    "    gene_start = start(all_genes)[gene_hits],\n",
    "    gene_end = end(all_genes)[gene_hits],\n",
    "    gene_strand = as.character(strand(all_genes)[gene_hits]),\n",
    "    stringsAsFactors = FALSE\n",
    "  )\n",
    "  \n",
    "  # Calculate distances\n",
    "  results$distance_to_gene <- abs(results$variant_pos - results$gene_start)\n",
    "  \n",
    "  # Filter by window size (double-check)\n",
    "  results <- results[results$distance_to_gene <= window_size, ]\n",
    "  \n",
    "  cat(\"Found\", nrow(results), \"variant-gene pairs\\n\")\n",
    "  \n",
    "  return(results)\n",
    "}\n",
    "\n",
    "# Optimized function for very large datasets using data.table\n",
    "find_genes_batch_fast <- function(variants_df, chr_col = \"chr\", pos_col = \"pos\", \n",
    "                                 id_col = \"variant_id\", window_size = 500000) {\n",
    "  \n",
    "  # Convert to data.table for speed\n",
    "  variants_dt <- data.table(variants_df)\n",
    "  setnames(variants_dt, c(chr_col, pos_col, id_col), c(\"chr\", \"pos\", \"variant_id\"))\n",
    "  \n",
    "  # Convert genes to data.table\n",
    "  genes_dt <- data.table(\n",
    "    gene_id = mcols(all_genes)$gene_id,\n",
    "    chr = as.character(seqnames(all_genes)),\n",
    "    start = start(all_genes),\n",
    "    end = end(all_genes),\n",
    "    strand = as.character(strand(all_genes))\n",
    "  )\n",
    "  \n",
    "  # Set keys for efficient joining\n",
    "  setkey(variants_dt, chr)\n",
    "  setkey(genes_dt, chr)\n",
    "  \n",
    "  # Join by chromosome first\n",
    "  joined <- genes_dt[variants_dt, on = \"chr\", allow.cartesian = TRUE]\n",
    "  \n",
    "  # Calculate distances and filter\n",
    "  joined[, distance := abs(start - pos)]\n",
    "  nearby <- joined[distance <= window_size]\n",
    "  \n",
    "  # Select relevant columns\n",
    "  results <- nearby[, .(variant_id, variant_chr = chr, variant_pos = pos,\n",
    "                       gene_id, gene_chr = chr, gene_start = start, \n",
    "                       gene_end = end, gene_strand = strand, \n",
    "                       distance_to_gene = distance)]\n",
    "  \n",
    "  return(as.data.frame(results))\n",
    "}\n",
    "\n",
    "# Modified function to add gene symbols AND ENSG IDs\n",
    "add_gene_annotations <- function(results_df) {\n",
    "  \n",
    "  cat(\"Adding gene symbols and ENSG IDs...\\n\")\n",
    "  \n",
    "  # Get unique gene IDs\n",
    "  unique_genes <- unique(results_df$gene_id)\n",
    "  \n",
    "  # Map to symbols and ENSG IDs\n",
    "  gene_annotations <- select(org.Hs.eg.db, \n",
    "                            keys = unique_genes,\n",
    "                            columns = c(\"SYMBOL\", \"GENENAME\", \"ENSEMBL\"),\n",
    "                            keytype = \"ENTREZID\")\n",
    "  \n",
    "  # Rename ENSEMBL column to ENSG_ID for clarity\n",
    "  names(gene_annotations)[names(gene_annotations) == \"ENSEMBL\"] <- \"ENSG_ID\"\n",
    "  \n",
    "  # Merge back\n",
    "  results_with_annotations <- merge(results_df, gene_annotations, \n",
    "                                   by.x = \"gene_id\", by.y = \"ENTREZID\", \n",
    "                                   all.x = TRUE)\n",
    "  \n",
    "  # Handle cases where multiple ENSG IDs exist for one gene\n",
    "  # Keep the first one and add a column indicating if there were multiple\n",
    "  if (any(duplicated(results_with_annotations[, c(\"variant_id\", \"gene_id\")]))) {\n",
    "    cat(\"Warning: Some genes have multiple ENSG IDs. Taking the first one.\\n\")\n",
    "    results_with_annotations <- results_with_annotations[\n",
    "      !duplicated(results_with_annotations[, c(\"variant_id\", \"gene_id\")]), ]\n",
    "  }\n",
    "  \n",
    "  return(results_with_annotations)\n",
    "}\n",
    "\n",
    "# Enhanced function using EnsDb for direct ENSG mapping\n",
    "add_gene_annotations_ensdb <- function(results_df) {\n",
    "  \n",
    "  cat(\"Adding gene symbols and ENSG IDs using EnsDb...\\n\")\n",
    "  \n",
    "  # Check if EnsDb package is available\n",
    "  if (!require(\"EnsDb.Hsapiens.v86\", quietly = TRUE)) {\n",
    "    cat(\"EnsDb.Hsapiens.v86 not available, falling back to org.Hs.eg.db method\\n\")\n",
    "    return(add_gene_annotations(results_df))\n",
    "  }\n",
    "  \n",
    "  ensdb <- EnsDb.Hsapiens.v86\n",
    "  \n",
    "  # Get unique gene IDs\n",
    "  unique_genes <- unique(results_df$gene_id)\n",
    "  \n",
    "  # First map Entrez to ENSG using org.Hs.eg.db\n",
    "  entrez_to_ensg <- select(org.Hs.eg.db, \n",
    "                          keys = unique_genes,\n",
    "                          columns = c(\"ENSEMBL\"),\n",
    "                          keytype = \"ENTREZID\")\n",
    "  \n",
    "  # Then get gene symbols and names from EnsDb using ENSG IDs\n",
    "  ensg_ids <- unique(entrez_to_ensg$ENSEMBL[!is.na(entrez_to_ensg$ENSEMBL)])\n",
    "  \n",
    "  if (length(ensg_ids) > 0) {\n",
    "    ensg_symbols <- select(ensdb,\n",
    "                          keys = ensg_ids,\n",
    "                          columns = c(\"SYMBOL\", \"GENENAME\"),\n",
    "                          keytype = \"GENEID\")\n",
    "    \n",
    "    # Merge the mappings\n",
    "    gene_annotations <- merge(entrez_to_ensg, ensg_symbols, \n",
    "                             by.x = \"ENSEMBL\", by.y = \"GENEID\", \n",
    "                             all.x = TRUE)\n",
    "    \n",
    "    # Rename columns for clarity\n",
    "    names(gene_annotations)[names(gene_annotations) == \"ENSEMBL\"] <- \"ENSG_ID\"\n",
    "    \n",
    "    # Merge back to results\n",
    "    results_with_annotations <- merge(results_df, gene_annotations, \n",
    "                                     by.x = \"gene_id\", by.y = \"ENTREZID\", \n",
    "                                     all.x = TRUE)\n",
    "  } else {\n",
    "    # Fallback to org.Hs.eg.db only\n",
    "    results_with_annotations <- add_gene_annotations(results_df)\n",
    "  }\n",
    "  \n",
    "  # Handle duplicates (some genes may have multiple ENSG IDs)\n",
    "  if (any(duplicated(results_with_annotations[, c(\"variant_id\", \"gene_id\")]))) {\n",
    "    cat(\"Warning: Some genes have multiple ENSG IDs. Taking the first one.\\n\")\n",
    "    results_with_annotations <- results_with_annotations[\n",
    "      !duplicated(results_with_annotations[, c(\"variant_id\", \"gene_id\")]), ]\n",
    "  }\n",
    "  \n",
    "  return(results_with_annotations)\n",
    "}\n",
    "\n",
    "# Function to process in chunks (for memory efficiency)\n",
    "process_variants_chunked <- function(variants_df, chunk_size = 5000, \n",
    "                                   window_size = 500000) {\n",
    "  \n",
    "  n_variants <- nrow(variants_df)\n",
    "  n_chunks <- ceiling(n_variants / chunk_size)\n",
    "  \n",
    "  cat(\"Processing\", n_variants, \"variants in\", n_chunks, \"chunks\\n\")\n",
    "  \n",
    "  results_list <- list()\n",
    "  \n",
    "  for (i in 1:n_chunks) {\n",
    "    start_idx <- (i - 1) * chunk_size + 1\n",
    "    end_idx <- min(i * chunk_size, n_variants)\n",
    "    \n",
    "    cat(\"Processing chunk\", i, \"of\", n_chunks, \n",
    "        \"(variants\", start_idx, \"to\", end_idx, \")\\n\")\n",
    "    \n",
    "    chunk_variants <- variants_df[start_idx:end_idx, ]\n",
    "    chunk_results <- find_genes_batch_fast(chunk_variants, window_size = window_size)\n",
    "    \n",
    "    results_list[[i]] <- chunk_results\n",
    "  }\n",
    "  \n",
    "  # Combine all results\n",
    "  all_results <- rbindlist(results_list)\n",
    "  \n",
    "  return(as.data.frame(all_results))\n",
    "}\n",
    "\n",
    "# Example usage with sample data\n",
    "create_sample_variants <- function(n = 20000) {\n",
    "  \n",
    "  set.seed(123)\n",
    "  \n",
    "  # Create sample variant data\n",
    "  variants <- data.frame(\n",
    "    variant_id = paste0(\"var_\", 1:n),\n",
    "    chr = paste0(\"chr\", sample(1:22, n, replace = TRUE)),\n",
    "    pos = sample(1000000:200000000, n, replace = TRUE),\n",
    "    ref = sample(c(\"A\", \"T\", \"G\", \"C\"), n, replace = TRUE),\n",
    "    alt = sample(c(\"A\", \"T\", \"G\", \"C\"), n, replace = TRUE),\n",
    "    stringsAsFactors = FALSE\n",
    "  )\n",
    "  \n",
    "  return(variants)\n",
    "}\n",
    "\n",
    "# Performance comparison function\n",
    "benchmark_methods <- function(n_variants = 1000) {\n",
    "  \n",
    "  # Create test data\n",
    "  test_variants <- create_sample_variants(n_variants)\n",
    "  \n",
    "  # Method 1: Standard approach\n",
    "  cat(\"Testing standard method...\\n\")\n",
    "  time1 <- system.time({\n",
    "    results1 <- find_genes_batch(test_variants)\n",
    "  })\n",
    "  \n",
    "  # Method 2: Fast approach\n",
    "  cat(\"Testing fast method...\\n\")\n",
    "  time2 <- system.time({\n",
    "    results2 <- find_genes_batch_fast(test_variants)\n",
    "  })\n",
    "  \n",
    "  cat(\"Standard method:\", time1[3], \"seconds\\n\")\n",
    "  cat(\"Fast method:\", time2[3], \"seconds\\n\")\n",
    "  cat(\"Speedup:\", round(time1[3] / time2[3], 2), \"x\\n\")\n",
    "  \n",
    "  return(list(standard = results1, fast = results2))\n",
    "}\n",
    "\n",
    "# Main analysis function with ENSG ID support\n",
    "run_variant_analysis <- function(variants_file = NULL, variants_df = NULL, \n",
    "                                window_size = 500000, add_annotations = TRUE,\n",
    "                                use_ensdb = NULL,  # Auto-detect if NULL\n",
    "                                output_file = \"variant_gene_results.tsv\") {\n",
    "  \n",
    "  # Auto-detect EnsDb availability if not specified\n",
    "  if (is.null(use_ensdb)) {\n",
    "    use_ensdb <- require(\"EnsDb.Hsapiens.v86\", quietly = TRUE)\n",
    "    if (use_ensdb) {\n",
    "      cat(\"EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\\n\")\n",
    "    }\n",
    "  }\n",
    "  \n",
    "  # Load variants\n",
    "  if (!is.null(variants_file)) {\n",
    "    cat(\"Loading variants from\", variants_file, \"...\\n\")\n",
    "    variants_df <- fread(variants_file)\n",
    "  } else if (is.null(variants_df)) {\n",
    "    cat(\"Creating sample data...\\n\")\n",
    "    variants_df <- create_sample_variants(20000)\n",
    "  }\n",
    "  \n",
    "  # Process variants\n",
    "  if (nrow(variants_df) > 10000) {\n",
    "    # Use chunked processing for large datasets\n",
    "    results <- process_variants_chunked(variants_df, chunk_size = 5000, \n",
    "                                       window_size = window_size)\n",
    "  } else {\n",
    "    # Use fast method for smaller datasets\n",
    "    results <- find_genes_batch_fast(variants_df, window_size = window_size)\n",
    "  }\n",
    "  \n",
    "  # Add gene annotations (symbols and ENSG IDs) if requested\n",
    "  if (add_annotations) {\n",
    "    if (use_ensdb) {\n",
    "      results <- add_gene_annotations_ensdb(results)\n",
    "    } else {\n",
    "      results <- add_gene_annotations(results)\n",
    "    }\n",
    "  }\n",
    "  \n",
    "  # Reorder columns for better readability\n",
    "  if (add_annotations) {\n",
    "    column_order <- c(\"variant_id\", \"variant_chr\", \"variant_pos\",\n",
    "                     \"gene_id\", \"ENSG_ID\", \"SYMBOL\", \"GENENAME\",\n",
    "                     \"gene_chr\", \"gene_start\", \"gene_end\", \"gene_strand\",\n",
    "                     \"distance_to_gene\")\n",
    "    \n",
    "    # Only include columns that exist\n",
    "    existing_cols <- intersect(column_order, names(results))\n",
    "    results <- results[, existing_cols]\n",
    "  }\n",
    "  \n",
    "  # Save results\n",
    "  fwrite(results, output_file, sep = '\\t')\n",
    "  cat(\"Results saved to\", output_file, \"\\n\")\n",
    "  \n",
    "  # Summary statistics\n",
    "  cat(\"\\nSummary:\\n\")\n",
    "  cat(\"Total variant-gene pairs:\", nrow(results), \"\\n\")\n",
    "  cat(\"Unique variants with nearby genes:\", length(unique(results$variant_id)), \"\\n\")\n",
    "  cat(\"Unique genes found:\", length(unique(results$gene_id)), \"\\n\")\n",
    "  \n",
    "  if (add_annotations) {\n",
    "    cat(\"Genes with ENSG IDs:\", sum(!is.na(results$ENSG_ID)), \"\\n\")\n",
    "    cat(\"Genes with symbols:\", sum(!is.na(results$SYMBOL)), \"\\n\")\n",
    "  }\n",
    "  \n",
    "  return(results)\n",
    "}\n",
    "\n",
    "# Quick start examples:\n",
    "\n",
    "# Example 1: Process sample data with ENSG IDs (auto-detects EnsDb)\n",
    "# results <- run_variant_analysis()\n",
    "\n",
    "# Example 2: Process your own data with ENSG IDs\n",
    "# results <- run_variant_analysis(variants_file = \"my_variants.csv\")\n",
    "\n",
    "# Example 3: Process existing data frame with ENSG IDs\n",
    "# my_variants <- data.frame(...)\n",
    "# results <- run_variant_analysis(variants_df = my_variants)\n",
    "\n",
    "# Example 4: Different window size with ENSG IDs\n",
    "# results <- run_variant_analysis(window_size = 1000000)  # 1Mb window\n",
    "\n",
    "# Example 5: Force use of org.Hs.eg.db method only\n",
    "# results <- run_variant_analysis(use_ensdb = FALSE)\n",
    "\n",
    "# Example 6: Skip annotations (faster processing)\n",
    "# results <- run_variant_analysis(add_annotations = FALSE)\n",
    "\n",
    "cat(\"Script loaded successfully!\\n\")\n",
    "cat(\"Use run_variant_analysis() to process your variants.\\n\")\n",
    "cat(\"The output TSV file will now include ENSG_ID column.\\n\")\n",
    "cat(\"EnsDb.Hsapiens.v86 detected - enhanced ENSG mapping will be used by default.\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "1a021cd8-dc22-48f2-85d3-6f6950743fde",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: Some genes have multiple ENSG IDs. Taking the first one.\n",
      "Results saved to TL1_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 101907 \n",
      "Unique variants with nearby genes: 5567 \n",
      "Unique genes found: 3882 \n",
      "Genes with ENSG IDs: 90326 \n",
      "Genes with symbols: 86238 \n"
     ]
    }
   ],
   "source": [
    "TL1 <- run_variant_analysis(variants_df = TL1, output_file = \"TL1_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "41f52963-c196-4b70-8d5b-2a8b35bbf1d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:1 mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to mvage_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 8752 \n",
      "Unique variants with nearby genes: 777 \n",
      "Unique genes found: 250 \n",
      "Genes with ENSG IDs: 7668 \n",
      "Genes with symbols: 7458 \n"
     ]
    }
   ],
   "source": [
    "mvAge <- run_variant_analysis(variants_df = mvAge, output_file = \"mvage_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "e5baa46d-cd5e-4093-8305-d0fff422569a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: Some genes have multiple ENSG IDs. Taking the first one.\n",
      "Results saved to AD_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 59708 \n",
      "Unique variants with nearby genes: 5530 \n",
      "Unique genes found: 1573 \n",
      "Genes with ENSG IDs: 52842 \n",
      "Genes with symbols: 48595 \n"
     ]
    }
   ],
   "source": [
    "AD <- run_variant_analysis(variants_df = AD, output_file = \"AD_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "280fa730-26b2-4c99-b3c2-6311da655d9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: Some genes have multiple ENSG IDs. Taking the first one.\n",
      "Results saved to Aging_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 106967 \n",
      "Unique variants with nearby genes: 6054 \n",
      "Unique genes found: 4028 \n",
      "Genes with ENSG IDs: 94492 \n",
      "Genes with symbols: 90055 \n"
     ]
    }
   ],
   "source": [
    "Aging <- run_variant_analysis(variants_df = Aging, output_file = \"Aging_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "8e772057-79ac-4df9-b8f2-0a0797815e8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: Some genes have multiple ENSG IDs. Taking the first one.\n",
      "Results saved to PD_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 14907 \n",
      "Unique variants with nearby genes: 2041 \n",
      "Unique genes found: 381 \n",
      "Genes with ENSG IDs: 11537 \n",
      "Genes with symbols: 10411 \n"
     ]
    }
   ],
   "source": [
    "PD <- run_variant_analysis(variants_df = PD, output_file = \"PD_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "5367f08b-cce2-450c-8af5-28aa76344fad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EnsDb.Hsapiens.v86 detected - using enhanced ENSG mapping\n",
      "Processing 14008 variants in 3 chunks\n",
      "Processing chunk 1 of 3 (variants 1 to 5000 )\n",
      "Processing chunk 2 of 3 (variants 5001 to 10000 )\n",
      "Processing chunk 3 of 3 (variants 10001 to 14008 )\n",
      "Adding gene symbols and ENSG IDs using EnsDb...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: Some genes have multiple ENSG IDs. Taking the first one.\n",
      "Results saved to DNE_proxy_genes.tsv \n",
      "\n",
      "Summary:\n",
      "Total variant-gene pairs: 189309 \n",
      "Unique variants with nearby genes: 13925 \n",
      "Unique genes found: 3448 \n",
      "Genes with ENSG IDs: 163045 \n",
      "Genes with symbols: 154025 \n"
     ]
    }
   ],
   "source": [
    "DNE <- run_variant_analysis(variants_df = DNE, output_file = \"DNE_proxy_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e70d4d2-d015-444c-8f4f-cdb4442f7518",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
